pkg/pool-utils/contracts/controllers/ManagedPoolController.sol

```diff
@@ -0,0 +1,198 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+pragma solidity ^0.7.0;
+pragma experimental ABIEncoderV2;
+
+import "../interfaces/IControlledManagedPool.sol";
+import "./BasePoolController.sol";
+
+/**
+ * @dev Pool controller that serves as the "owner" of a Managed pool, and is in turn owned by
+ * an account empowered to make calls on this contract, which are forwarded to the underlyling pool.
+ *
+ * This contract can place limits on whether and how these calls can be made. For instance,
+ * imposing a minimum gradual weight change duration.
+ *
+ * While Balancer pool owners are immutable, ownership of this pool controller can be transferrable,
+ * if the corresponding permission is set.
+ */
+contract ManagedPoolController is BasePoolController, IControlledManagedPool {
+    using WordCodec for bytes32;
+
+    // There are five managed pool rights: all corresponding to permissioned functions of ManagedPool.
+    struct ManagedPoolRights {
+        bool canChangeWeights;
+        bool canDisableSwaps;
+        bool canSetMustAllowlistLPs;
+        bool canSetCircuitBreakers;
+        bool canChangeTokens;
+    }
```

Comment:
Do we want to move these to the interface? Potentially not as `IControlledManagedPool` is generic to all `ManagedPool` controllers right?

Reply:
Yes, I'm inclined to leave them in the contracts, where the coupling is a lot tighter. You could theoretically have a right that isn't associated with a function.

We could also move the BasePoolRights - but the BasePoolController just has the initialize function.

Reply:
Cool, just wanted to float the topic.